pkg/apis/core/types.go

```diff
@@ -3553,17 +3575,32 @@ type ServiceSpec struct {
 	// More info: https://kubernetes.io/docs/concepts/services-networking/service/
 	Selector map[string]string
 
-	// ClusterIP is the IP address of the service and is usually assigned
-	// randomly by the master. If an address is specified manually and is not in
-	// use by others, it will be allocated to the service; otherwise, creation
-	// of the service will fail. This field can not be changed through updates.
-	// Valid values are "None", empty string (""), or a valid IP address. "None"
-	// can be specified for headless services when proxying is not required.
-	// Only applies to types ClusterIP, NodePort, and LoadBalancer. Ignored if
-	// type is ExternalName.
-	// More info: https://kubernetes.io/docs/concepts/services-networking/service/#virtual-ips-and-service-proxies
+	// ClusterIPs identifies all the ClusterIPs assigned to or requested for  this
+	// service. ClusterIPs are assigned or reserved  based on the values of
+	// Service.Spec.IPFamilies. Maximum of two entries (dualstack IPs) are
+	// allowed in ClusterIPs. The IPFamily of each ClusterIP must match
```

Comment:
is goal == > 2 IPs some day  @andrewsykim ?
if *yes* so make that explicit  in the comments, otherwise id prefer (ClusterIP4, ClusterIP6) ?

Reply:
hmm, not sure i understand you. Can you please elaborate?

Reply:
I don't think there was any plan to ever have more than 2 Cluster IPs (unlike with, say, PodIPs, where we did discuss that possibility), but from experience so far in porting stuff from single-stack to dual-stack, it generally ends up being easier to work with an array than with separate fields.